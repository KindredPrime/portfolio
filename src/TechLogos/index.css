.TechLogos img {
  margin-right: 5%;
}

.TechLogos__html {
  width: 55px;
}

.TechLogos__css {
  width: 55px;
}

.TechLogos__html, .TechLogos__css {
  /*
    These apply to alt text when the image doesn't load
  */
  color: black;
  font-size: 14px;
}

.TechLogos__jquery {
  width: 50px;
}

.TechLogos__react {
  width: 60px;
}

.TechLogos__node {
  width: 70px;
  padding: 4px;
  background-color: white;
}

.TechLogos__postgresql {
  width: 70px;
  background-color: white;
  padding: 3px;
}

.TechLogos__node, .TechLogos__postgresql {
  margin-top: 15px;
}

/*
  Tablet devices
*/
@media screen and (min-width: 690px) {
  .TechLogos__jquery {
    width: 70px;
  }

  .TechLogos__node, .TechLogos__postgresql {
    margin-top: 0px;
  }
}