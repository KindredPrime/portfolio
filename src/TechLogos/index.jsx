import PropTypes from 'prop-types';
import './index.css';

function TechLogos(props) {
  const { logos } = props;

  return (
    <div className="TechLogos">
      {logos.map(({id, src, alt, classNames}) => (
        <img key={`logo-${id}`} src={src} alt={alt} className={classNames} />
      ))}
    </div>
  );
}

TechLogos.propTypes = {
  logos: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      src: PropTypes.string.isRequired,
      alt: PropTypes.string.isRequired,
      classNames: PropTypes.string.isRequired
    })
  ).isRequired
}

export default TechLogos;