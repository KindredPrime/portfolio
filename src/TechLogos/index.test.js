import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import testLogo from '../images/testing/test.png';
import TechLogos from './index';

describe('TechLogos Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <TechLogos
        logos={[
          {
            id: 1,
            src: '',
            alt: '',
            classNames: ''
          }
        ]}
      />,
      div
    );
  });

  it('renders as expected', () => {
    render(
      <TechLogos
        logos={[
          {
            id: 1,
            src: testLogo,
            alt: 'Test alt',
            classNames: 'Test class names'
          }
        ]}
      />
    );

    expect(document.body).toMatchSnapshot();
  });
});