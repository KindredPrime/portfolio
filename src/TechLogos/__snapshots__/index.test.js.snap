// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechLogos Component renders as expected 1`] = `
<body>
  <div>
    <div
      class="TechLogos"
    >
      <img
        alt="Test alt"
        class="Test class names"
        src="test.png"
      />
    </div>
  </div>
</body>
`;
